src/plugins/fcrypt/fcrypt.c

```diff
@@ -235,20 +235,55 @@ static int fcryptGpgCallAndCleanup (Key * parentKey, KeySet * pluginConfig, char
 		if (rename (tmpFile, keyString (parentKey)) != 0)
 		{
 			// if rename failed we can still try to copy the file content manually
-			lseek (tmpFileFd, 0, SEEK_SET);
-			lseek (parentKeyFd, 0, SEEK_SET);
+			if (lseek (tmpFileFd, 0, SEEK_SET))
+			{
+				ELEKTRA_SET_RESOURCE_ERRORF (
```

Comment:
Why not a single "ELEKTRA_SET_RESOURCE_ERRORF" at the very end? (errno must be saved, though)